frontend/src/shared/metadata/instance-type-dropdown.tsx

```diff
@@ -40,7 +43,98 @@ export type InstanceTypeSelectorProperties = {
         | 'ProductionVariantInstanceType';
 };
 
+type InstanceTypeMultiSelectorProperties = InstanceTypeProperties & {
+    selectedOptions: OptionDefinition[]; 
+};
+
+type InstanceTypeSelectorProperties = InstanceTypeProperties & {
+    selectedOption: OptionDefinition | null;
+    service?: string;
+};
+
+function getInstanceTypes (computeTypes: string[]) {
+    const categorizedTypes: Record<string, OptionDefinition[]> = {
+        Standard: [],
+        'Compute Optimized': [],
+        'Memory Optimized': [],
+        'Accelerated computing': [],
+    };
+
+    computeTypes.forEach(
+        (instanceType) => {
+            // Instance types returned from this call are in the format of "ml.<family>.size".
+            // We can categorize these instances based on the first letter of the
+            // <family> value.
+            let category = 'Accelerated computing';
+            switch (instanceType.split('.')[1][0]) {
+                case 't':
+                case 'm':
+                    category = 'Standard';
+                    break;
+                case 'c':
+                    category = 'Compute Optimized';
+                    break;
+                case 'r':
+                    category = 'Memory Optimized';
+                    break;
+            }
```

Comment:
Could use a map for this which is technically more efficient (O=1):
```
categoryMap = {
  'm' : 'Standard',
  ...
}
```

```
const categoryLetter = instanceType.split('.')[1][0]);
if (categoryMap.has(categoryLetter){
  category = categoryMap[categoryLetter];
} else {
  category = "Accelerated computing"; // Should be general?
}
```

Also should we be using these family names? https://docs.aws.amazon.com/ec2/latest/instancetypes/instance-type-names.html

Reply:
I'm re-using the existing code we had for determining instance families; the families look correct when they're rendered but I think Dustin put this together originally so he might have more insight. 

I'm not seeing how a map is more efficient, the switch case is executed in constant time too?